client/src/components/ProfileForm.js

```diff
@@ -74,15 +109,41 @@ export default function ProfileForm({ onSubmit, image, upload, signIn }) {
   function validateForm(event) {
     const form = event.target.parentElement.parentElement
     const inputFullName = form.elements.fullName.value.trim()
+    const inputEmail = form.elements.email.value.trim()
+    const inputPassword = form.elements.password.value.trim()
     const inputAboutYou = form.elements.aboutYou.value.trim()
-    setIsActive(inputFullName && inputAboutYou ? false : true)
+    setIsActive(
+      !inputFullName || !inputAboutYou || !inputEmail || !inputPassword
+    )
+  }
+
+  function upload(event) {
+    const url = `https://api.cloudinary.com/v1_1/${CLOUDNAME}/upload`
+
+    const formData = new FormData()
+    formData.append('file', event.target.files[0])
+    formData.append('upload_preset', PRESET)
+
+    axios
+      .post(url, formData, {
+        headers: {
+          'Content-type': 'multipart/form-data',
+        },
+      })
+      .then(onImageSave)
+      .catch(err => console.error(err))
+  }
+
+  function onImageSave(response) {
+    setImage(response.data.url)
   }
```

Comment:
It is only a small suggestion and a personal preference, it is totally fine to do it the way you did though:
```suggestion
      .then(res => setImage(res.data.url))
      .catch(err => console.error(err))
  }
```

Reply:
Super, it works!